hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -1071,14 +1096,29 @@ CompletableFuture<FlushRegionResponse> flushRegionInternal(byte[] regionName, by
           .completeExceptionally(new NoServerForRegionException(Bytes.toStringBinary(regionName)));
         return;
       }
-      addListener(flush(serverName, location.getRegion(), columnFamily, writeFlushWALMarker),
-        (ret, err2) -> {
-          if (err2 != null) {
-            future.completeExceptionally(err2);
-          } else {
-            future.complete(ret);
-          }
-        });
+      TableName tableName = location.getRegion().getTable();
+      addListener(getDescriptor(tableName), (tDesc, error2) -> {
```

Comment:
See above about introducing the additional RPC.

IMHO, the client should be dumber, not smarter. I'd rather see the `NoSuchColumnFamilyException` come from the RS than the client.

Reply:
Agree. I think an NoSuchColumnFamilyException here would be better too.